quinn-proto/src/connection/mod.rs

```diff
@@ -1176,23 +1176,29 @@ where
             return;
         }
 
-        let (space, count) = self.pto_time_and_space(now).map_or_else(
-            // Handshake anti-deadlock packet
-            || {
-                debug_assert!(self.side.is_client() && self.highest_space <= SpaceId::Handshake);
-                (self.highest_space, 1)
-            },
-            // PTO
-            |(_, pto_space)| (pto_space, 2),
-        );
-
+        let (_, space) = match self.pto_time_and_space(now) {
+            Some(x) => x,
+            None => {
+                error!("PTO expired while unset");
+                return;
+            }
+        };
         trace!(
             in_flight = self.in_flight.bytes,
             count = self.pto_count,
             ?space,
             "PTO fired"
         );
 
+        let count = if self.in_flight.ack_eliciting == 0 {
+            // A PTO when we're not expecting any ACKs must be due to handshake anti-amplification
+            // deadlock preventions
+            debug_assert!(!self.peer_completed_address_validation());
+            1
+        } else {
+            // Conventional loss probe
+            2
+        };
```

Comment:
I think this might be easier to follow when written as `match self.in_flight.ack_eliciting {}`?